### PR TITLE
KUI 1140 change semester options

### PR DESCRIPTION
**Pull request by amirhossein-haerian**
This PR is part of this task: https://kth-se.atlassian.net/browse/KUI-1140 
There is also another PR related to this task which is in the kurs-pm-admin-web repository with same branch name.
If you have any comments regarding the place of created files (or you think there is a better place for them) please share that as well.

### CODE REVIEW

**Review by karlandindrakryggen — CHANGES_REQUESTED** [review]
Looks good, but I would like you to look at and think about if `SemesterDropdown` needs an internal state.

**Review by karlandindrakryggen — APPROVED** [review]

**Review by axelbjo — APPROVED** [review]

File: public/js/app/components/SemesterDropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          semesterList.map(({ term }) => (
+            <option data-testid="select-option" id={`itemFor-${term}`} key={term} value={term}>
+              {seasonStr(langIndex, term)}
+            </option>
+          ))}
+      </select>
+    </div>
+  )
+}
+export default memo(SemesterDropdown)
```
Comment (belanglos) [inline public/js/app/components/SemesterDropdown.jsx:35]: Same comment as in the other repo. I think we do not need `memo` here.

File: public/js/app/pages/CreateNewMemo.jsx [hunk 2]
```diff
@@ -108,12 +108,11 @@ function CreateNewMemo(props) {
...
 
-  const onChoiceOfSemester = event => {
-    const newSemester = event.target.value
+  const onChoiceOfSemester = useCallback(selectedSemester => {
     const { existingDraftEndPoint } = chosen
-    setSemester(newSemester)
+    setSemester(selectedSemester)
     _cleanUpPrevCheckboxesState(existingDraftEndPoint)
-  }
+  }, [])
```
Comment (belanglos) [inline public/js/app/pages/CreateNewMemo.jsx:115]: Is eslint complaining about `_cleanUpPrevCheckboxesState` missing from the dep-array?

File: test/unit/helper.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    expect(roundIsNotOutdated(date)).toBe(false)
+  })
+
+  it('should return true for the last day of the previous year', () => {
+    const lastYear = new Date().getFullYear() - 1
+    const date = new Date(`${lastYear}-12-31`).toISOString()
+    expect(roundIsNotOutdated(date)).toBe(true)
+  })
+
+  it('should return error for an invalid date', () => {
```
Comment (belanglos) [inline test/unit/helper.test.js:27]: 🏅

**Review by belanglos — CHANGES_REQUESTED** [review]
Good stuff!
Similar comments as in the other repo, please hit me up if you have questions :)

**Review by belanglos — APPROVED** [review]
Great job!!

**Pull request merged**